dns/resolver.py

```diff
@@ -447,13 +447,13 @@ def __init__(self, key, value):
         self.prev = self
```

Comment:
For my next MR:

This class is currently a bit problematic. The "sentinel" object is instantiated as `LRUCacheNode(None, None)`, implying that the arguments are optional. But then `get()` etc of the cache thing that key/value might be None, throwing lots of type errors.

A solution would be to introduce a `SentinelCacheNode` class that allows None values, or to `# type: ignore` the instantiation of the sentinel node.

What do you think?

Reply:
I don't think I'd change this, as the LRUCacheNode is really internal to the implementation and is not exposed.  Maybe just an assert that node.value is not None if needed?

Reply:
Hmm, I just reread and see I missed part of your point.  I'll ponder this a bit more.  I'm also ok with just leaving this class untyped as it's basically internal.  (In retrospect, I should have called it _LRUCacheNode).  But I'll think about your idea a little more too.